app/controllers/application_controller.rb

```diff
@@ -15,6 +15,11 @@ class ApplicationController < ActionController::Base
   before_action :set_paper_trail_whodunnit
   before_action :no_cache
 
+  rescue_from CanCan::AccessDenied do |e|
+    sign_out
```

Comment:
I don't know if you should sign out the user if they aren't on the whitelist. This would force users who aren't on the VPN to basically have to go through the annoying sign in workflow again

Reply:
@corymcdonald unfortunately they go into a redirect loop if they're signed in. That being said, we should increase the sign in limits as it's currently too restrictive.